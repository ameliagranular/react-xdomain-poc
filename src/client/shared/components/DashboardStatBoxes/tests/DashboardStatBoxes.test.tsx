import 'jest'; // necessary for types

import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';

import {
    DashboardStatBoxes,
    IDashboardStatBoxesProps
} from '../DashboardStatBoxes';

// Configure Enzyme for use with Jest & ES6+
configure({ adapter: new Adapter() });


const wrapper = (props: IDashboardStatBoxesProps = {}) => (
    <DashboardStatBoxes {...props} />
);

const doMount = (props: IDashboardStatBoxesProps = {}) => {
    return mount(wrapper(props));
};

describe('DashboardStatBoxes', () => {
    it('renders expected markup with no props', () => {
        expect(toJson(doMount())).toMatchSnapshot();
    });

    it('renders expected markup with pending', () => {
        expect(toJson(doMount({
            pending: true
        }))).toMatchSnapshot();
    });

    it('renders expected markup with pending and count', () => {
        expect(toJson(doMount({
            pending: true,
            count: 2
        }))).toMatchSnapshot();
    });

    it('renders expected markup with data', () => {
        expect(toJson(doMount({
            elements: [
                { val: <span>$1,000</span>, label: <span>Amount Lost</span> },
                { val: <span>$4,000</span>, label: <span>Amount Gained</span> }
            ]
        }))).toMatchSnapshot();
    });
});
