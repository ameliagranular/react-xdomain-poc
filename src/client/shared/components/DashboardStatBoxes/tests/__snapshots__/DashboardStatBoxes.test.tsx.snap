// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardStatBoxes renders expected markup with data 1`] = `
<DashboardStatBoxes
  count={1}
  elements={
    Array [
      Object {
        "label": <span>
          Amount Lost
    </span>,
        "val": <span>
          $1,000
    </span>,
      },
      Object {
        "label": <span>
          Amount Gained
    </span>,
        "val": <span>
          $4,000
    </span>,
      },
    ]
  }
  pending={false}
>
  <div
    className="DashboardStatBoxes DashboardStatBoxesReady"
  >
    <StatBox
      key="0"
      label={
        <span>
          Amount Lost
        </span>
      }
      val={
        <span>
          $1,000
        </span>
      }
    >
      <div
        className="DashboardStatBoxes_StatBox"
      >
        <div
          className="DashboardStatBoxes_StatBox_Value"
        >
          <span>
            $1,000
          </span>
        </div>
        <div
          className="DashboardStatBoxes_StatBox_Label"
        >
          <span>
            Amount Lost
          </span>
        </div>
      </div>
    </StatBox>
    <StatBox
      key="1"
      label={
        <span>
          Amount Gained
        </span>
      }
      val={
        <span>
          $4,000
        </span>
      }
    >
      <div
        className="DashboardStatBoxes_StatBox"
      >
        <div
          className="DashboardStatBoxes_StatBox_Value"
        >
          <span>
            $4,000
          </span>
        </div>
        <div
          className="DashboardStatBoxes_StatBox_Label"
        >
          <span>
            Amount Gained
          </span>
        </div>
      </div>
    </StatBox>
  </div>
</DashboardStatBoxes>
`;

exports[`DashboardStatBoxes renders expected markup with no props 1`] = `
<DashboardStatBoxes
  count={1}
  elements={null}
  pending={false}
>
  <div
    className="DashboardStatBoxes DashboardStatBoxesPending"
  >
    <StatBox
      key="0"
      label={
        <span>
          --
        </span>
      }
      val={
        <span>
          --
        </span>
      }
    >
      <div
        className="DashboardStatBoxes_StatBox"
      >
        <div
          className="DashboardStatBoxes_StatBox_Value"
        >
          <span>
            --
          </span>
        </div>
        <div
          className="DashboardStatBoxes_StatBox_Label"
        >
          <span>
            --
          </span>
        </div>
      </div>
    </StatBox>
  </div>
</DashboardStatBoxes>
`;

exports[`DashboardStatBoxes renders expected markup with pending 1`] = `
<DashboardStatBoxes
  count={1}
  elements={null}
  pending={true}
>
  <div
    className="DashboardStatBoxes DashboardStatBoxesPending"
  >
    <StatBox
      key="0"
      label={
        <span>
          --
        </span>
      }
      val={
        <span>
          --
        </span>
      }
    >
      <div
        className="DashboardStatBoxes_StatBox"
      >
        <div
          className="DashboardStatBoxes_StatBox_Value"
        >
          <span>
            --
          </span>
        </div>
        <div
          className="DashboardStatBoxes_StatBox_Label"
        >
          <span>
            --
          </span>
        </div>
      </div>
    </StatBox>
  </div>
</DashboardStatBoxes>
`;

exports[`DashboardStatBoxes renders expected markup with pending and count 1`] = `
<DashboardStatBoxes
  count={2}
  elements={null}
  pending={true}
>
  <div
    className="DashboardStatBoxes DashboardStatBoxesPending"
  >
    <StatBox
      key="0"
      label={
        <span>
          --
        </span>
      }
      val={
        <span>
          --
        </span>
      }
    >
      <div
        className="DashboardStatBoxes_StatBox"
      >
        <div
          className="DashboardStatBoxes_StatBox_Value"
        >
          <span>
            --
          </span>
        </div>
        <div
          className="DashboardStatBoxes_StatBox_Label"
        >
          <span>
            --
          </span>
        </div>
      </div>
    </StatBox>
    <StatBox
      key="1"
      label={
        <span>
          --
        </span>
      }
      val={
        <span>
          --
        </span>
      }
    >
      <div
        className="DashboardStatBoxes_StatBox"
      >
        <div
          className="DashboardStatBoxes_StatBox_Value"
        >
          <span>
            --
          </span>
        </div>
        <div
          className="DashboardStatBoxes_StatBox_Label"
        >
          <span>
            --
          </span>
        </div>
      </div>
    </StatBox>
  </div>
</DashboardStatBoxes>
`;
